include/myst/thread.h

```diff
@@ -29,6 +29,13 @@
 /* Typical default value in host operating system's /proc/sys/kernel/pid_max */
 #define MYST_PID_MAX 0x8000
 
+/*
+ * Bitmap to track pids in use. If bit 'n' (starting from 0) is set, then pid
+ * 'n' is assigned to an active process or thread. When cycling, we cycling pid
+ * between [104, 32677].
+ */
+MYST_UNUSED static uint8_t bitmap_pid[4096] = {0};
```

Comment:
this is wrong. This should be a global variable in something like thread.c itself, not defined in the header. This is adding this arrary to every source file that includes thread.h which is cauing a lot of bloat.
By all means make this an extern variable in the header, but only define it once.